src/apps/core/app.js

```diff
@@ -89,13 +90,35 @@ define(function(require) {
 			self.displayFavicon();
 			self.loadSVG();
 
+			if (crossSiteMessaging !== undefined) {
+				self.bindCrossSiteMessagingHandler(crossSiteMessaging);
+			}
+
 			container.append(mainTemplate);
 
 			monster.waterfall([
 				_.bind(self.loadIncludes, self)
 			], _.bind(self.loadAuth, self));
 		},
 
+		bindCrossSiteMessagingHandler: function(crossSiteMessaging) {
+			const { origin, topics } = crossSiteMessaging;
+
+			window.addEventListener('message', handler);
+
+			function handler(event) {
+				if (event.origin !== origin || !topics.includes(event.data)) {
+					return;
+				}
+
+				if (!monster.util.isAuthorizedTopicForCrossSiteMessaging(event.data)) {
+					return;
+				}
+
+				monster.pub('core.crossSiteMessage', event.data);
```

Comment:
Could there be a way to route the messages for each app, also taking into account if it is active, or not? Or maybe even adding the name of the app to the data, so for example we could call:

```suggestion
				monster.pub(event.data.app + '.crossSiteMessage', event.topic);
```

Should we also consider allowing to send some payload to the application, besides the topic?

Reply:
Apologies for forgetting to include a check for the active/load status of the application. It has been fixed now. After reviewing the .pub and .sub documentation, I decided to use 'core.crossSiteMessage' along with the application name.

We could, but I'm really not sure, as this was just an enhancement based on @joristirado's RFC to the previous implementation. Moreover, Verizon themselves will not be using this anymore.